pkg/alertmanager/alertmanager_metrics_test.go

```diff
@@ -257,6 +257,226 @@ func TestAlertmanagerMetricsStore(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestAlertmanagerMetricsRemoval(t *testing.T) {
+	mainReg := prometheus.NewPedanticRegistry()
+
+	alertmanagerMetrics := newAlertmanagerMetrics()
+	mainReg.MustRegister(alertmanagerMetrics)
+	alertmanagerMetrics.addUserRegistry("user1", populateAlertmanager(1))
+	alertmanagerMetrics.addUserRegistry("user2", populateAlertmanager(10))
+	alertmanagerMetrics.addUserRegistry("user3", populateAlertmanager(100))
+
+	alertmanagerMetrics.removeUserRegistry("user3")
```

Comment:
I'd suggest adding a test before removal and after removal. Make sure that counters don't go down -- that would be interpreted as a reset by Prometheus.